### DISCUSSION TITLE
Reload on SIGHUP

### DISCUSSION THREAD

**Discussion by sirex** [Kludex/uvicorn#1869] [Ideas]
It would be useful to have a possibility to reload uvicorn manually with SIGHUP signal.

> [Daemon](https://en.wikipedia.org/wiki/Daemon_(computing)) programs sometimes use SIGHUP as a signal to restart themselves, the most common reason for this being to re-read a configuration file that has been changed.
> -- https://en.wikipedia.org/wiki/SIGHUP

For example if I change some configuration files, I would like to do:

    $ kill -HUP $PID

Now terminate uvicorn and start it again.

**Comment by T-101** [comment]
+1 for this, this is the way I do things with gunicorn. It's a delightfully easy way to hotreload latest commits to a dockerized application by just saying `docker exec app kill -HUP 1`

**Comment by vvanglro** [comment]
The new process manager already supports this feature. https://github.com/encode/uvicorn/pull/2183

You can update uvicorn to >=0.30.0.

```sh
kill -SIGHUP $PID
```

**Comment by uwu-420** [comment]
@sirex @T-101 am I missing something or isn't this a rather damaging idea? When I close the terminal where the app is running the app gets a SIGHUP. So I'd actually expect uvicorn to do a graceful app shutdown in this case just like it does for SIGINT and SIGTERM. Having the app crash badly just because I close the terminal isn't what I'd consider sensible at all.

**Reply by T-101** [reply]
Turns out I was able to trigger the hotreload with kill -2 (SIGINT?), so I'm a happy camper

**Reply by uwu-420** [reply]
Isn't that quite unintuitive? I'm so confused :D

**Reply by T-101** [reply]
It is what it is. That feature allows me to trigger a code reload without bringing the container down.
